codalab/worker_manager/azure_batch_worker_manager.py

```diff
@@ -0,0 +1,152 @@
+        subparser.add_argument(
+            '--account-name', type=str, help='Azure Batch account name',
+        )
+        subparser.add_argument(
+            '--account-key', type=str, help='Azure Batch account key',
+        )
+        subparser.add_argument(
+            '--service-url', type=str, help='Azure Batch service URL',
+        )
+        subparser.add_argument(
+            '--log-container-url',
+            type=str,
+            help='URL of the Azure Storage container to store the worker logs',
+        )
+        subparser.add_argument(
+            '--job-id', type=str, help='ID of the Azure Batch job to add tasks to',
+        )
+        subparser.add_argument(
+            '--cpus', type=int, default=1, help='Default number of CPUs for each worker'
+        )
+        subparser.add_argument(
+            '--gpus', type=int, default=0, help='Default number of GPUs to request for each worker'
+        )
+        subparser.add_argument(
+            '--memory-mb', type=int, default=2048, help='Default memory (in MB) for each worker'
+        )
+        subparser.add_argument(
+            '--user', type=str, default='root', help='User to run the Batch jobs as'
+        )
+
+    def __init__(self, args):
+        super().__init__(args)
+
+        credentials: SharedKeyCredentials = SharedKeyCredentials(
+            self.args.account_name, self.args.account_key
+        )
+        self._batch_client: BatchServiceClient = BatchServiceClient(
+            credentials, batch_url=self.args.service_url
+        )
+        self._batch_client.config.retry_policy.retries = 1
+
+    def get_worker_jobs(self) -> List[WorkerJob]:
+        # Count the number active and running tasks for the Azure Batch job
+        task_counts: TaskCounts = self._batch_client.job.get_task_counts(self.args.job_id)
+        return [WorkerJob(True) for _ in range(task_counts.active + task_counts.running)]
+
+    def start_worker_job(self) -> None:
+        worker_image: str = 'codalab/worker:' + os.environ.get('CODALAB_VERSION', 'latest')
+        worker_id: str = uuid.uuid4().hex
+        logger.debug('Starting worker {} with image {}'.format(worker_id, worker_image))
+        work_dir_prefix: str = (
+            self.args.worker_work_dir_prefix if self.args.worker_work_dir_prefix else "/tmp/"
+        )
+
+        # This needs to be a unique directory since Batch jobs may share a host
+        work_dir: str = os.path.join(work_dir_prefix, 'cl_worker_{}_work_dir'.format(worker_id))
+        command: List[str] = self.build_command(worker_id, work_dir)
+
+        task_container_run_options: List[str] = [
+            '--cpus %d' % self.args.cpus,
+            '--memory %dM' % self.args.memory_mb,
+            '--volume /var/run/docker.sock:/var/run/docker.sock',
+            '--volume %s:%s' % (work_dir, work_dir),
+            '--user %s' % self.args.user,
+        ]
+
+        if os.environ.get('CODALAB_USERNAME') and os.environ.get('CODALAB_PASSWORD'):
```

Comment:
If this `if` block doesn't run, the worker won't be able to connect, right? Can we make it error if these environment variables aren't set?
(The AWSBatchWorkerManager errors because job definition creation fails in the case that these variables are None, but it could probably be more explicit there as well)